scrapers/scraper.py

```diff
@@ -0,0 +1,268 @@
+
+            if any(commitData["sha"] == cm["sha"] for cm in history):
+                continue
+
+            dixFile = "https://raw.githubusercontent.com/apertium/apertium-{0}/{1}/apertium-{0}.{0}.dix".format(pairName, commitData["sha"])
+            stems = countDixStems(dixFile, True)
+            if stems == -1:
+                dixName = iso3to2[pairList[0]] + "-" + iso3to2[pairList[1]]
+                dixFile = "https://raw.githubusercontent.com/apertium/apertium-{0}/{2}/apertium-{1}.{1}.dix".format(pairName, dixName, commitData["sha"])
+                stems = countDixStems(dixFile, True)
+                if stems == -1:
+                    continue
+
+            commitData["stems"] = stems["stems"]
+            history.append(commitData)
+
+        langPackages.append({"name": pairName, "history": history})
+    return langPackages
+
+def monoData(packages, updatemailmap):
+    """Returns data for all monolingual dictionaries: state, stems, location and contributors"""
+    data = []
+    for package in packages:
+        if not(re.match(r"apertium-\w+$", package["name"]) and package["name"][len("apertium-"):] in languages):
+            continue
+
+        language = package["name"][len("apertium-"):]
+        stats = json.loads(requests.get("https://apertium.projectjj.com/stats-service/apertium-{}/Lexc".format(language)).text)["stats"]
+        for statistic in stats:
+            if statistic["stat_kind"] == "Stems":
+                stems = statistic["value"]
+                break
+        for topic in package["topics"]:
+            if topic[len("apertium-"):] in langLocations:
+                location = topic[len("apertium-"):]
+                break
+
+        subprocess.call('git clone https://github.com/apertium/apertium-{}'.format(language), stderr=subprocess.DEVNULL)
+        shutil.copyfile("./.mailmap", "./apertium-{}/.mailmap".format(language))
+        lines = subprocess.check_output('git log --format="%aN" --follow apertium-{0}.{0}.lexc'.format(language), cwd='./apertium-{}/'.format(language)).decode('utf-8')
+
+        if updatemailmap:
+            emails = subprocess.check_output('git log --format="<%aE>" --follow apertium-{0}.{0}.lexc'.format(language), cwd='./apertium-{}/'.format(language)).decode('utf-8').split("\n")
+            mailmap = open("./.mailmap", "r").read()
+            for email in emails:
+                if not email in mailmap:
+                    print(email)
+
+        shutil.rmtree("./apertium-{}/".format(language), onerror=del_rw)
+
+        authors = lines.split("\n")
+        authors.pop() #last line is always empty
+        contributors = []
+        authorCount = Counter(authors)
+        for contributor, count in authorCount.items():
+            contributors.append({"user": contributor, "value": count})
+
+        wikiData = requests.get(wikiURL).text
+        rows = etree.fromstring(wikiData)[1][0][0][0][3][3][8]
```

Comment:
This like gives me `xml.etree.ElementTree.ParseError: not well-formed (invalid token): line 5, column 18`

Reply:
I can't replicate this error...

Reply:
[Here's the error I get](https://imgur.com/a/7auYbMY), even with the updated code. I'm running this on Ubuntu 18.10. Also, could you change all the double quotes to single quotes or vice versa for consistency please?

Reply:
Tested it on a VM with Ubuntu 18 and still can’t replicate the error. I did run the code after all of my recent changes, so maybe I accidentally fixed it?